src/Microsoft.Extensions.DependencyInjection/ServiceProvider.cs

```diff
@@ -17,18 +17,19 @@ namespace Microsoft.Extensions.DependencyInjection
     /// </summary>
     internal class ServiceProvider : IServiceProvider, IDisposable
     {
-        private readonly ServiceProvider _root;
         private readonly ServiceTable _table;
         private bool _disposeCalled;
-
-        private readonly Dictionary<IService, object> _resolvedServices = new Dictionary<IService, object>();
         private List<IDisposable> _transientDisposables;
 
+        internal ServiceProvider Root { get; }
```

Comment:
I don't have strong opinion on this, `internal` property in `internal` class seems to be pointless now when I look at it.